packages/template-retail-react-app/app/hooks/use-shopper-context-search-params.js

```diff
@@ -0,0 +1,135 @@
+    const queryClient = useQueryClient()
+    const {search} = useLocation()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {parameters: {usid, siteId}},
+        {enabled: !isServer}
+    )
+    console.log('shopperContext', shopperContext)
+
+    const refetchDataOnClient = () => {
+        queryClient.invalidateQueries()
+    }
+
+    useEffect(() => {
+        const updateShopperContextObj = getShopperContextSearchParams(search)
+        if (Object.keys(updateShopperContextObj).length > 0) {
+            handleShopperContextUpdate({
+                createShopperContext,
+                updateShopperContext,
+                shopperContext,
+                updateShopperContextObj,
+                usid,
+                siteId,
+                refetchDataOnClient
+            })
+        }
+    }, [search])
+
+    useEffect(() => {
+        console.log('isHydrated()', isHydrated())
+        if (isHydrated()) {
+            refetchDataOnClient()
+        }
+    }, [])
+}
+
+export const getShopperContextSearchParams = (
+    search,
+    arraySearchParams = SHOPPER_CONTEXT_ARRAY_FIELDS
+) => {
+    const searchParamsObj = new URLSearchParams(search)
+    const getSearchParam = (key, isList = false) =>
+        searchParamsObj.has(key)
+            ? {[key]: isList ? searchParamsObj.getAll(key) : searchParamsObj.get(key)}
+            : {}
+
+    const reduceParams = (keys) =>
+        keys.reduce(
+            (acc, key) => ({...acc, ...getSearchParam(key, arraySearchParams.includes(key))}),
+            {}
+        )
+
+    const shopperContextSearchParams = {
+        ...(searchParamsObj.has(SHOPPER_CONTEXT_SEARCH_PARAMS.SOURCE_CODE) && {
+            sourceCode: searchParamsObj.get(SHOPPER_CONTEXT_SEARCH_PARAMS.SOURCE_CODE)
+        }),
+        ...(searchParamsObj.has(SHOPPER_CONTEXT_SEARCH_PARAMS.EFFECTIVE_DATE_TIME) && {
+            effectiveDateTime: searchParamsObj.get(
+                SHOPPER_CONTEXT_SEARCH_PARAMS.EFFECTIVE_DATE_TIME
+            )
+        }),
+        ...(searchParamsObj.has(SHOPPER_CONTEXT_SEARCH_PARAMS.CUSTOMER_GROUP_IDS) && {
+            customerGroupIds: searchParamsObj.getAll(
+                SHOPPER_CONTEXT_SEARCH_PARAMS.CUSTOMER_GROUP_IDS
+            )
+        })
+    }
+    const customQualifiers = reduceParams(
+        Object.values(SHOPPER_CONTEXT_CUSTOM_QUALIFIERS_SEARCH_PARAMS)
+    )
+    const assignmentQualifiers = reduceParams(
+        Object.values(SHOPPER_CONTEXT_ASSIGNMENT_QUALIFIERS_SEARCH_PARAMS)
+    )
+
+    return {
+        ...shopperContextSearchParams,
+        ...(Object.keys(customQualifiers).length && {customQualifiers}),
+        ...(Object.keys(assignmentQualifiers).length && {assignmentQualifiers})
+    }
+}
+
+export const handleShopperContextUpdate = async ({
+    createShopperContext,
```

Comment:
It is a bit uncommon to pass the actions into another function like this. Let's bring the logic in this func into `useShopperContextSearchParams`.